src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -871,6 +871,8 @@ private fun processNestedScopesUpwards(
         if (processLexicalDeclarations(scope, cameFrom, ns, withPlainExternCrateItems, shadowingProcessor)) return@walkUp true
         prevScope.addAll(currScope)
         false
+    }) {
+        return false
```

Comment:
Hm, looks like we should return `true` here because a successful result of `if (walkUp())` means that `processor` found suitable item and whole call of `processNestedScopesUpwards` is successful.
Otherwise, all callee would consider that they should continue to search suitable items